### PR TITLE
feat: use street address templates from locales

### PR DESCRIPTION
**Pull request by ST-DDT**
Adds #88

* [x] Adds `faker.address.buildingNumber()`
* [x] Rewrite `faker.address.streetAddress()` to use the templates from the locales
* [x] Update locale template files to the `fake()` format

### CODE REVIEW

**Comment by ST-DDT** [comment]
Depends on #756

**Comment by ST-DDT** [comment]
#756 was merged

**Comment by ST-DDT** [comment]
I converted the street_address templates using the following script:

````ts
function updateLocaleFileHook(
  filePath: string,
  locale: string,
  localePath: string[]
): void {
  if (localePath[0] === 'address' && localePath[1] === 'street_address') {
    console.log(filePath + ' <-> ' + locale + ' @ ' + localePath.join(' -> '));
    let content: string[] = require(filePath).default.map((entry) =>
      entry
        .replace('#{street_name}', '{{address.streetName}}')
        .replace('#{address.street_name}', '{{address.streetName}}')
        .replace('#{building_number}', '{{address.buildingNumber}}')
        .replace('#{address.building_number}', '{{address.buildingNumber}}')
        .replace('#{secondary_address}', '{{address.secondaryAddress}}')
    );
    content = [
      content[0].replace(' {{address.secondaryAddress}}', '').replace(/,$/, ''),
      content[1] ??
        content[0].replace(' {{address.secondaryAddress}}', '') +
          ' {{address.secondaryAddress}}',
    ];

    let fileContent =
      'export default ' + JSON.stringify(content) + ' as [string, string]';
    fileContent = format(fileContent, prettierTsOptions);
    console.log(fileContent);
    writeFileSync(filePath, fileContent);
  }
}
````

During the implementation I noticed, that some locale files contained two patterns, one with the normal pattern and one with the full pattern.
Since there seems to be a special pattern to these to, I decided to change the implementation again to use only the format from the template.

I'm not sure whether `[string, string]` is the best type, I also considered `{ normal: string; full: string }`.
Which do you like the most?

File: src/locales/fa/address/street_address.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 export default [
-  '#{street_name}, پلاک #{building_number}',
-  '#{street_name}, #{street_name}, پلاک #{building_number}',
-];
+  '{{address.streetName}}, پلاک {{address.buildingNumber}}',
+  '{{address.streetName}}, {{address.streetName}}, پلاک {{address.buildingNumber}}',
```
Comment (ST-DDT) [inline src/locales/fa/address/street_address.ts:3]:
```suggestion
  '{{address.streetName}}, پلاک {{address.buildingNumber}}  {{address.secondaryAddress}}',
```

**Comment by ST-DDT** [comment]
<details>
<summary><strong>Here are some examples for the generated addresses: (click to expand)</strong></summary>



| Locale | Normal | Full |
| --- | --- | --- |
| af_ZA | 1027 Dippenaar Neck | 574 Schmidt Motorway Suite 833 |
| ar | 90492 جاوحدو Oval | 4506 بندر Roads 246 شقة |
| az | Ethelyn pr., 417 | King sahəsi, 739 m. 453 |
| cz | Foltýnová Lights 831 | Ctibor Neck 0 Apt. 248 |
| de | Flore Pass 9 | Estelle Loaf 30b Apt. 526 |
| de_AT | Ruckdeschel Centers 06 | Linus Canyon 58 Zimmer 554 |
| de_CH | 7344 Schwab Burg | 343 Sidler Wells Suite 718 |
| el | 1214 Στέφανος Stravenue | 0082 Ελευθέριος Burg Suite 445 |
| en | 0261 Boehm Port | 637 Raleigh Causeway Suite 971 |
| en_AU | 74 Moss Track | 38 Matilda Terrace Apt. 165 |
| en_AU_ocker | 19 Nigel Crescent | 9997 Samuel Loop Suite 283 |
| en_BORK | 5677 Eveline View | 9131 Johnson Burg Suite 826 |
| en_CA | 839 Langworth Flat | 3834 Walker Ville Apt. 608 |
| en_GB | 803 Denesik Wall | 9847 Baumbach Valley Suite 893 |
| en_GH | 511 Asamoa Street | 26 Agyare Road Apt. 225 |
| en_IE | 73197 Johanna Stream | 36351 Cruickshank Flat Apt. 603 |
| en_IND | 7453 Guneta Views | 0986 Gudakesha Plaza Suite 227 |
| en_NG | 948 Shalewa Squares | 65637 Chukwu Harbors Apt. 708 |
| en_US | 7225 Erick Lane | 712 Lakin Turnpike Suite 071 |
| en_ZA | 001 Rafael Junctions | 9731 Jonathan Knolls Suite 730 |
| es | Guerrero Huerta s/n. | Juan Carlos Calleja 6 Puerta 594 |
| es_MX | Arellano Partida 0 | Ana Luisa Monte s/n. Edificio 1 |
| fa | علی‌پور آفریقا, پلاک 19 | همایون کارگر شمالی, پلاک 36  9 طبقه |
| fi | 4464 Nieminen Park | 81229 Annikki Estates Suite 564 |
| fr | 8879 Mathieu du Chat-qui-Pêche | 617 Leclerc Montorgueil Apt. 149 |
| fr_BE | Meryl de Barchon 935c | Tasnim du Chat-qui-Pêche 057b Suite 780 |
| fr_CA | 875 Enola Tunnel | 520 Loyce Extensions Suite 221 |
| fr_CH | 163 Augustine Via | 841 Magdalena Circle Suite 115 |
| ge | ჩიხრაძე გამზირი 89 | აბაშიძე ხეივანი 49 კორპ. 93 |
| he | 221 באר Lakes | 785 בןגוריון Shore 45 חדר |
| hr | Ancica Stream 2 | Ada Inlet 4 Stan 71 |
| hy | Նարեկ  փողոց 614/4 | Գևորգ  պողոտա 38 Բն. 298 |
| id_ID | Kiehn Street no 84 | Madyson Spur no 18 Apt. 983 |
| it | Daciano Via 22 | Alcide Rotonda 8, Appartamento 74 |
| ja | 2459 結愛 Roads | 949 太一 Glens Apt. 519 |
| ko | 4750 늘찬 면 | 3014 준빈 읍 Apt. 210 |
| lv | Jonathon gāte 21 | Javier laukums 192 - 808 |
| mk | Чаловска Mountain 3 | Бужаровски Isle 068 стан 42 |
| nb_NO | Mathisen hagen 5 | Leah holtet 7 Leil. 640 |
| ne | 83445 Hamal Mountain | 83133 Ajita Canyon Suite 014 |
| nl | Dijk dijk 7 | Berg laan 916 II 3 hoog |
| nl_BE | Lotte park 5 | Luna plein 1 3e verdieping |
| pl | Apollinary Mews 074 | Klaudiusz Plain 8730 Apt. 562 |
| pt_BR | 729 Braga Travessa | 74314 Barros Avenida Lote 48 |
| pt_PT | Renato Camp 974 | Madalena Course 3 Apt. 817 |
| ro | Elton Intrarea, Bloc 79 | Aimee Intrarea, Bloc 81, Ap. 695 |
| ru | Bridget проспект, 000 | Garret улица, 622 кв. 979 |
| sk | Bernhard Haven 4 | Hallie Orchard 8 Suite 713 |
| sv | Klara gatan 3 | Karl vägen 4 Lgh. 617 |
| tr | Okumuş Summit 53b | Orbay Crossroad 23a Apt. 764 |
| uk | Johnson майдан, 3 | Earline майдан, 53 кв. 791 |
| ur | 7901 نیازی کالونی | 783 نیازی آباد گلی 844 |
| vi | 0023 Đào Loaf | 01215 Quang Triệu Harbor Apt. 530 |
| zh_CN | 龚 街54号 | 赖 桥37号 Apt. 372 |
| zh_TW | 昊強 西路15號 | 子涵 東路6325號 Apt. 443 |
| zu_ZA | 73476 Bayeni Union | 601 Mbende Underpass Suite 035 |

</details>

**Comment by pkuczynski** [comment]
> Here are some examples for the generated addresses

Polish one (`pl`) is not Polish at all regarding both format and language. How can I improve that?

File: src/locales/ar/address/street_address.ts [hunk 2]
```diff
@@ -1 +1,4 @@
-export default ['#{building_number} #{street_name}'];
+export default [
+  '{{address.buildingNumber}} {{address.streetName}}',
+  '{{address.buildingNumber}} {{address.streetName}} {{address.secondaryAddress}}',
+] as [string, string];
```
Comment (pkuczynski) [inline src/locales/ar/address/street_address.ts:4]: You don't really need to cast it, do you? Type can be derived from values...

Comment (ST-DDT) [reply src/locales/ar/address/street_address.ts:4]: Sadly this doesnt work for tuples.

**Comment by ST-DDT** [comment]
> > Here are some examples for the generated addresses
> 
> Polish one (`pl`) is not Polish at all regarding both format and language. How can I improve that?

I could change/fix the format here, but it might be best if you create a separate PR for that (format/street address, street name, building number and secondary address)

File: src/locales/ar/address/street_address.ts (hunk 2 above)
Comment (clarkerican) [reply src/locales/ar/address/street_address.ts:4]: I like the idea of using an object here. It'll make the code a lot more readable

**Review by Shinigami92 — APPROVED** [review]

**Review by pkuczynski — APPROVED** [review]

**Pull request merged**